drivers/bluetooth/hci/ipm_stm32wb.c

```diff
@@ -139,57 +144,71 @@ static void tryfix_event(TL_Evt_t *tev)
 
 void TM_EvtReceivedCb(TL_EvtPacket_t *hcievt)
 {
-	struct net_buf *buf;
-	struct bt_hci_acl_hdr acl_hdr;
-	TL_AclDataSerial_t *acl;
-
-	k_sem_take(&ipm_busy, K_NO_WAIT);
-
-	switch (hcievt->evtserial.type) {
-	case HCI_EVT:
-		BT_DBG("EVT: hcievt->evtserial.evt.evtcode: 0x%02x",
-		       hcievt->evtserial.evt.evtcode);
-		switch (hcievt->evtserial.evt.evtcode) {
-		case BT_HCI_EVT_VENDOR:
-			/* Vendor events are currently unsupported */
-			BT_ERR("Unknown evtcode type 0x%02x",
+	k_fifo_put(&ipm_rx_events_fifo, hcievt);
+}
+
+static void bt_ipm_rx_thread(void)
+{
+	while (true) {
+		static TL_EvtPacket_t *hcievt;
+		struct net_buf *buf = NULL;
+		struct bt_hci_acl_hdr acl_hdr;
+		TL_AclDataSerial_t *acl;
+
+		hcievt = k_fifo_get(&ipm_rx_events_fifo, K_FOREVER);
+
+		k_sem_take(&ipm_busy, K_NO_WAIT);
```

Comment:
Not really related to this PR, but what is the ipm_busy used for? Also there is no check or handling of the return value.

Reply:
`ipm_busy` is used to avoid simultaneous TX/RX. Need to check about the NO_WAIT though...

Reply:
@joerchan Thanks for pointing this. This was meant to be a mutex protecting access to shared memory. 
The `NO_WAIT` made it no-op.
I'll check if this should be fixed or removed... and push a seperate PR.